### PR TITLE
fixed: appear bug on quickly updating components

### PR DESCRIPTION
**Pull request by jquense**

### CODE REVIEW

File: src/Transition.js [hunk 1]
```diff
@@ -258,10 +258,19 @@ class Transition extends React.Component {
   }
 
   safeSetState(nextState, callback) {
+    // We need to track pending updates for instances where a cWRP fires quickly
```
Comment (taion) [inline src/Transition.js]: Is this a React 16 thing? How can a cWRP fire before cDM flushes?

Comment (jquense) [reply src/Transition.js]: its happening on v15 for me. I don't know. I'd imagine that's not behavior you want but it definitely seems to be happening

File: src/Transition.js [hunk 2]
```diff
@@ -258,10 +258,19 @@ class Transition extends React.Component {
...
+    this.pendingState = nextState;
+
     // This shouldn't be necessary, but there are weird race conditions with
     // setState callbacks and unmounting in testing, so always make sure that
     // we can cancel any pending setState callbacks after we unmount.
-    this.setState(nextState, this.setNextCallback(callback));
+    callback = this.setNextCallback(callback)
+    this.setState(nextState, () => {
+      this.pendingState = null;
+      callback()
```
Comment (taion) [inline src/Transition.js]: semi

File: src/Transition.js [hunk 3]
```diff
@@ -258,10 +258,19 @@ class Transition extends React.Component {
...
+    // We need to track pending updates for instances where a cWRP fires quickly
+    // after cDM and before the state flushes, which would double trigger a
+    // transition
+    this.pendingState = nextState;
+
     // This shouldn't be necessary, but there are weird race conditions with
     // setState callbacks and unmounting in testing, so always make sure that
     // we can cancel any pending setState callbacks after we unmount.
-    this.setState(nextState, this.setNextCallback(callback));
+    callback = this.setNextCallback(callback)
```
Comment (taion) [inline src/Transition.js]: semi

File: src/Transition.js (hunk 1 above)
Comment (taion) [reply src/Transition.js]: weird

**Comment by nmOobert** [comment]
I think I am hitting this bug. Is there a way I can install this change? Or what is an expected release of this?

**Comment by STRML** [comment]
We hit the same. A v2.3.0 release would be very welcome so nobody else hits it.

**Comment by jquense** [comment]
working on a release right now

**Pull request merged**